app/repository/pg/user.go

```diff
@@ -32,20 +32,38 @@ func NewRepo(db *sql.DB) *Repo {
 // Add meth implements usecase.UserRepository interface
 // without even knowing it that allow us to decouple our layers
 // and will make our app flexible and maintainable.
-func (r Repo) Add(user entities.User) (string, error) {
+func (r Repo) AddUser(user entities.User) (string, error) {
 	var id string
 
 	const sqlStatement = `INSERT INTO "user" (first_name, last_name, email, password)
 						  VALUES ($1, $2, $3, $4) RETURNING id`
 
 	if err := r.DB.QueryRow(sqlStatement, user.FirstName, user.LastName, user.Email, hash.SHA256(user.Password)).Scan(&id); err != nil {
 		var pqErr = new(pq.Error)
-		if errors.As(err, &pqErr) && pqErr.Code.Name() == uniqueViolationErrCode {
-			return "", ErrDuplicateEmail
+		if errors.As(err, &pqErr) && pqErr.Code.Name() == globs.UniqueViolationErrCode {
+			return "", globs.ErrDuplicateEmail
 		}
 
 		return "", fmt.Errorf("error inserting into database: %w", err)
 	}
 
 	return id, nil
 }
+
+// FindUser method implements usecase.UserRepository logic
+// finding user in DB by ID.
+func (r Repo) FindUser(id string) (*entities.User, error) {
+	var user entities.User
+
+	const sqlStatement = `SELECT id, first_name, last_name, email, created_at FROM "user" WHERE id=$1`
+
+	if err := r.DB.QueryRow(sqlStatement, id).Scan(&user.ID, &user.FirstName, &user.LastName, &user.Email, &user.CreatedAt); err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Better to check with `errors.Is()` because e.g. you might add some intermediate layer and wrap there the `sql.ErrNoRows` value and then at this place the error will not be equal to the `sql.ErrNoRows` but will be a wrap of it.

Reply:
fixed